libflux/flux-core/src/parser/mod.rs

```diff
@@ -1488,23 +1494,28 @@ impl<'input> Parser<'input> {
             name: t.lit,
         }
     }
-    fn parse_int_literal(&mut self) -> IntegerLit {
+    fn parse_int_literal(&mut self) -> Result<IntegerLit, TokenError> {
         let t = self.expect(TokenType::Int);
+
+        if t.lit.starts_with('0') && t.lit.len() > 1 {
+            return Err(TokenError { token: t })
+        }
+
         match (&t.lit).parse::<i64>() {
             Err(_e) => {
                 self.errs.push(format!(
                     "invalid integer literal \"{}\": value out of range",
```

Comment:
Why do we return an error from this function instead of push the onto the `self.errs` vector like is already done here?

Reply:
Oh true, I was caught up adding a small guard and making it clear it could have an error result with the error token, similar to float and duration literal parsing, but stepping back, it looks like just pushing to the errs vector with a descriptive message would be more helpful too. I could push an err message like `"invalid integer literal \"{}\": leading zero not supported"` or something particular to the cause.